Mage.Sets/src/mage/cards/c/CallForAid.java

```diff
@@ -0,0 +1,127 @@
+package mage.cards.c;
+
+import mage.abilities.Ability;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.RestrictionEffect;
+import mage.abilities.effects.common.UntapAllEffect;
+import mage.abilities.effects.common.continuous.CantBeSacrificedSourceEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAllEffect;
+import mage.abilities.effects.common.continuous.GainControlAllEffect;
+import mage.abilities.keyword.HasteAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.PermanentInListPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetOpponent;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * @author spillnerdev
+ */
+public final class CallForAid extends CardImpl {
+
+    public CallForAid(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{4}{R}");
+
+        // Gain control of all creatures target opponent controls until end of turn. Untap those creatures. They gain haste until end of turn. You can't attack that player this turn. You can't sacrifice those creatures this turn.
+        getSpellAbility().addTarget(new TargetOpponent());
+        getSpellAbility().addEffect(new CallForAidEffect());
+    }
+
+    private CallForAid(final CallForAid card) {
+        super(card);
+    }
+
+    @Override
+    public CallForAid copy() {
+        return new CallForAid(this);
+    }
+}
+
+
+class CallForAidEffect extends OneShotEffect {
+
+    public CallForAidEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "Gain control of all creatures target opponent controls until end of turn. Untap those creatures. They gain haste until end of turn. You can't attack that player this turn. You can't sacrifice those creatures this turn.";
+    }
+
+    protected CallForAidEffect(CallForAidEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        UUID targetOpponent = getTargetPointer().getFirst(game, source);
+        if (targetOpponent != null) {
+            // First we get all creatures of target player...
+            FilterCreaturePermanent filter = new FilterCreaturePermanent();
+            List<Permanent> opponentCreatures = game.getBattlefield().getActivePermanents(filter, targetOpponent, game);
+            filter.add(new PermanentInListPredicate(opponentCreatures));
```

Comment:
I really need to deprecate this, it's not stable across game cycles and won't work here. Instead use `PermanentReferenceInCollectionPredicate`.